lib/engine/game/base.rb

```diff
@@ -104,18 +104,36 @@ class Base
       # with reason => after
       #   reason: What kind of game end check to do
       #   after: When game should end if check triggered
-      # Leave out a reason if game does not support that.
-      # Allowed reasons:
-      #  bankrupt, stock_market, bank, final_train, final_phase, all_closed, custom
+      # Reasons implemented here:
+      #   - all_closed: all corporations and companies have closed
+      #   - bank: bank is broken
+      #   - bankrupt
+      #   - custom: override to implement game-specifc reason
+      #   - final_phase: reached the last phase
+      #   - final_train: no trains left in the Depot
+      #   - stock_market: a corporation reached an end game cell on the stock
+      #                   market
       # Allowed after:
-      #  immediate - ends in current turn
+      #  immediate - ends after current action
       #  current_round - ends at the end of the current round
       #  current_or - ends at the next end of an OR
       #  full_or - ends at the next end of a complete OR set
       #  one_more_full_or_set - finish the current OR set, then
       #                         end after the next complete OR set
       GAME_END_CHECK = { bankrupt: :immediate, bank: :full_or }.freeze
 
+      GAME_END_TIMING_PRIORITY = {
```

Comment:
Why hash instead of an array where index is the priority?

Reply:
I personally tend to err on the side of maintainability when the performance gains are minimal. In this case, if I wanted the O(1), I'd construct the hash from the array at init time, giving the best of both worlds at the cost of a meager amount of memory.

Reply:
Changed to an array using `.index`